### PR TITLE
ci(cypress): Added Config Testcases

### PR DESCRIPTION
**Pull request by Gnanasundari24**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [x] CI/CD

## Description
 Added below config related testcases in cypress
  "collect_shipping_details_from_wallet_connector"
  "collect_billing_details_from_wallet_connector"
  "always_collect_billing_details_from_wallet_connector"
  "always_collect_shipping_details_from_wallet_connector"

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

To validate the config changes in automation

## How did you test it?
 Ran locally
 
 
Screenshot 2024-11-20 at 6 53 20 PM
Screenshot 2024-11-20 at 6 56 25 PM

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/fixtures/update-business-profile.json
Comment (pixincreate) [inline cypress-tests/cypress/fixtures/update-business-profile.json:1]: can we move all this into a single file? preferably `business-profile.json` and keep them all as objects like we've done in `v2`? that way you will not have the need to keep multiple files for one area viz., `business-profile`.

at present, we've 2 files namely,

- `create-business-profile.json
- `update-business-profile.json`

and both of these have at max 3 lines.

we can merge them into one and call them via objects in `.cy.js` files.

**Review by pixincreate — COMMENTED** [review]
minor nit picks.

**Review by likhinbopanna — APPROVED** [review]

File: cypress-tests/cypress/e2e/PaymentTest/00024-ConfigTest.cy.js [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+      });
+
+      it("Update collect_billing_details_from_wallet_connector to true", () => {
+        cy.UpdateBusinessProfileTest(
+          fixtures.updateBusinessProfile,
+          true, // is_connector_agnostic_enabled
+          true, // collect_billing_address_from_wallet_connector
+          false, //collect_shipping_address_from_wallet_connector
+          false, //always_collect_billing_address_from_wallet_connector
+          false, //always_collect_shipping_address_from_wallet_connector
```
[the remark this answers is not in the document]
Comment (pixincreate) [reply cypress-tests/cypress/e2e/PaymentTest/00024-ConfigTest.cy.js:57]: can you update this as well?

File: cypress-tests/cypress/e2e/PaymentTest/00024-ConfigTest.cy.js [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+          req_data,
+          res_data,
+          "no_three_ds",
+          "automatic",
+          globalState
+        );
+        if (should_continue)
+          should_continue = utils.should_continue_further(res_data);
+      });
+      it("payment_methods-call-test", () => {
```
[the remark this answers is not in the document]
Comment (pixincreate) [reply cypress-tests/cypress/e2e/PaymentTest/00024-ConfigTest.cy.js:79]: even this. there exist many more down the line.

File: cypress-tests/cypress/e2e/PaymentTest/00024-ConfigTest.cy.js [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+      });
+
+      it("Update collect_shipping_details_from_wallet_connector to true", () => {
+        cy.UpdateBusinessProfileTest(
+          fixtures.updateBusinessProfile,
+          true, // is_connector_agnostic_enabled
+          false, // collect_billing_address_from_wallet_connector
+          false, //collect_shipping_address_from_wallet_connector
+          false, //always_collect_billing_address_from_wallet_connector
+          false, //always_collect_shipping_address_from_wallet_connector
```
[the remark this answers is not in the document]
Comment (pixincreate) [reply cypress-tests/cypress/e2e/PaymentTest/00024-ConfigTest.cy.js:103]: this one as well

**Review by likhinbopanna — APPROVED** [review]

File: cypress-tests/cypress/e2e/PaymentTest/00024-ConfigTest.cy.js
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentTest/00024-ConfigTest.cy.js:1]: can you update the file name? we now have 2 files that starts with `00024`

**Review by pixincreate — APPROVED** [review]

**Review by likhinbopanna — APPROVED** [review]

**Pull request merged**